### PR TITLE
Fix the @with_deprecated decorator.

### PR DESCRIPTION
**Pull request by ghost**
### What does this PR do?

Fixes the `@with_deprecated decorator`, which at the moment (at least in the development branch) breaks.

The steps to reproduce this issue:

1. Add the following snipped at the end of salt/modules/apache.py:

```
from salt.utils.decorators import with_deprecated

@with_deprecated(globals(), "Fluorine")
def foo(bar='foobar', quiet=False):
     return True


def _foo(bar='foobar', quiet=False):
      return False

```

2. Run `salt-call apache.foo` and get:

```
Passed invalid arguments: coercing to Unicode: need string or buffer, function found.

Usage:
None
Traceback (most recent call last):
  File "/Volumes/User/sjl/work/salt/salt/salt/cli/caller.py", line 212, in call
    ret['return'] = func(*args, **kwargs)
  File "/Volumes/User/sjl/work/salt/salt/salt/utils/decorators/__init__.py", line 558, in _decorate
    return self._call_function(kwargs)
  File "/Volumes/User/sjl/work/salt/salt/salt/utils/decorators/__init__.py", line 252, in _call_function
    error = six.text_type(error).replace(self._function, self._orig_f_name)  # Hide hidden functions
TypeError: coercing to Unicode: need string or buffer, function found
```

The correct result should be instead the value True.

The problem seems to be stemming from the fact that `_DeprecationDecorator._get_args()` in `salt/utils/decorators/__init__.py` assumes module calls to have a key `__pub_arg` in `kwargs`, which is not the case.

### What does this PR do?

Replaces `_DeprecationDecorator._get_args()` in `salt/utils/decorators/__init__.py` with a function which calls `salt.utils.args.clean_kwargs` to remove other than function specific keywords (and their values) from `kwargs` (which will then be passed to `_call_function()` etc.).

### What issues does this PR fix or reference?

See the main description above. This is also indirectly related is #46554.

### Previous Behavior

Functions decorated with `@with_deprecated` produced an error (see above) when called with `salt-call mod.func`.

### New Behavior

Functions decorated with `@with_deprecated` work as expected when called with `salt-call mod.func`.

### Tests written?

No

### Commits signed with GPG?

No

### CODE REVIEW

**Comment by samilaine** [comment]
This is more or less a precision labeled with @isbm (indirectly related to #46554).

File: salt/utils/decorators/__init__.py [hunk 1]
```diff
@@ -216,22 +216,13 @@ def __init__(self, globals, version):
...
-
-        if '__pub_arg' in kwargs:  # For modules
-            for arg_item in kwargs.get('__pub_arg', list()):
-                if type(arg_item) == dict:
-                    _kwargs.update(arg_item.copy())
-                else:
-                    _args.append(arg_item)
-        else:
-            _kwargs = kwargs.copy()  # For states
+        _kwargs = salt.utils.args.clean_kwargs(**kwargs)
```
Comment (isbm) [inline salt/utils/decorators/__init__.py]: I am happy this is now finally an utility. :wink:

**Review by isbm — APPROVED** [review]
Thanks!

**Comment by rallytime** [comment]
re-run all

**Comment by samilaine** [comment]
I'm sorry to ask this, but I don't have a faintest idea what "re-run all" means. All I could do was to push "Update branch" -button, which I hope is what was asked for.

**Comment by isbm** [comment]
@samilaine this is just telling Jenkins to rerun all the tests and see if that breaks anything around. :wink:

**Review by cachedout — APPROVED** [review]
Much cleaner.

**Review by rallytime — APPROVED** [review]
Yes, much cleaner. 👍

**Comment by rallytime** [comment]
Thanks @samilaine!

**Pull request merged**